### PR TITLE
Random initialization for NUTS

### PR DESCRIPTION
**Pull request by fonnesbeck**
Added a `"random"` initialization option for NUTS that simply draws from the prior. Seems to work pretty well in some cases. Might be a good idea for initializing other samplers as well.

### CODE REVIEW

File: pymc3/examples/gelman_bioassay.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import pymc3 as pm
-from numpy import ones, array
+from pymc3 import *
```
Comment (twiecki) [inline pymc3/examples/gelman_bioassay.py]: I think we prohibit this now.

Comment (fonnesbeck) [reply pymc3/examples/gelman_bioassay.py]: Sorry, was testing and forgot to take it out.

**Comment by fonnesbeck** [comment]
Generalized to use with any step method

File: pymc3/distributions/discrete.py [hunk 2]
```diff
@@ -389,7 +389,7 @@ def _random(self, lower, upper, size=None):
         # as array-like.
         samples = stats.uniform.rvs(lower, upper - lower - np.finfo(float).eps,
                                     size=size)
-        return np.floor(samples).astype('int32')
+        return np.floor(samples).astype('int64')
```
Comment (twiecki) [inline pymc3/distributions/discrete.py]: Do we need int64 here? This causes problems with float32 interaction.

Comment (fonnesbeck) [reply pymc3/distributions/discrete.py]: I doubt it. I think I was trying to diagnose a bug here. I will remove it.

Comment (fonnesbeck) [reply pymc3/distributions/discrete.py]: `dtype='int64'` is the default for the `Discrete` class. Should that be changed?

**Comment by fonnesbeck** [comment]
I will never understand `git rebase` I fixed the conflict below 3 TIMES on my local machine. Time to turn on `rerere`, I guess.

**Comment by twiecki** [comment]
Sometimes it's better to make a new branch from master and cherry-pick the commits you want (can squash them before to make that faster).

**Comment by fonnesbeck** [comment]
Yes, I need to do that more. I am lazy.

File: pymc3/sampling.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from collections import defaultdict
+rom collections import defaultdict
```
Comment (AustinRochford) [inline pymc3/sampling.py]: Missing an `f`

Comment (fonnesbeck) [reply pymc3/sampling.py]: Yeah, fixed. Thanks.

**Comment by fonnesbeck** [comment]
Same rebase issue here. Closing.

**Pull request closed without merging**